src/renderer/services/wallets.js

```diff
@@ -73,31 +79,168 @@ export default {
   },
 
   openSavedWallet(name, passphrase) {
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
       try {
         const walletToOpen = this.getOne(name);
+        const net = network.getSelectedNetwork().net;
+
         const wif = wallet.decrypt(walletToOpen.encryptedWIF, passphrase);
         const account = new wallet.Account(wif);
+
+        let walletToOpenNew = storageNew.getOne(wif);
+        let mnemonic;
+        let encryptedMnemonicString;
+        let btcTestnetWalletClient;
+        let btcMainnetWalletClient;
+        let btcKey;
+
+        if (walletToOpen.encryptedMnemonicString === undefined) {
+          const dataBTCgen = await this.generateBTCETHWallet(name);
+          if (dataBTCgen.err) {
+            return reject('An error occured while trying to generate a new Bitcoin & Ethereum wallet.');
+          }
+          mnemonic = dataBTCgen.key.mnemonic;
+          btcKey = dataBTCgen.key;
+          btcTestnetWalletClient = dataBTCgen.btcTestnetWalletClient;
+          btcMainnetWalletClient = dataBTCgen.btcMainnetWalletClient;
+          const encryptedMnemonic = CryptoJS.AES.encrypt(mnemonic, account.WIF);
+          encryptedMnemonicString = encryptedMnemonic.toString();
+
+          this
+            .add(name, {
+              label: name,
+              encryptedWIF: walletToOpen.encryptedWIF,
+              address: walletToOpen.address,
+              scriptHash: walletToOpen.scriptHash,
+              encryptedMnemonicString,
+            })
+            .sync();
+
+          storageNew
+            .add(account.WIF, {
+              label: account.WIF,
+              encryptedMnemonicString,
+              isNew: true,
+            })
+            .sync();
+        } else {
+          const mnemonicBytes = CryptoJS.AES.decrypt(walletToOpen.encryptedMnemonicString, wif);
+          mnemonic = mnemonicBytes.toString(CryptoJS.enc.Utf8);
+
+          if (walletToOpenNew.isNew) {
+            const opts = {
+              name,
+              m: 1,
+              n: 1,
+              myName: 'me',
+              bwsurl: 'https://bws.bitpay.com/bws/api',
+              singleAddress: false,
+              coin: 'btc',
+            };
+            await this.createTestnetBTCWallet(walletToOpenNew.key, opts);
+            await this.createMainnetBTCWallet(walletToOpenNew.key, opts);
+          }
+
+          const dataBTCgen = await this.getBTCWalletObjects(mnemonic);
+          if (dataBTCgen.err) {
+            return reject('An error occured while trying to login to Bitcoin wallet.');
+          }
+          btcKey = dataBTCgen.key;
+          btcTestnetWalletClient = dataBTCgen.btcTestnetWalletClient;
+          btcMainnetWalletClient = dataBTCgen.btcMainnetWalletClient;
+        }
+        const seedString = bip39.mnemonicToSeedSync(mnemonic, wif).toString('hex');
```

Comment:
Why does this use the wif here?
Is this to try to make the use case of opening a wallet from a wif and a mneumonic seed work without needing to also specify a passphrase?

Reply:
we are encrypting mnemonics with wif, suppose user login through wif only, then what should happen? so to maintain this we encrypted the mnemonics with wif and store it.